pkg/auth/providers/azure/clients/ms_graph_client.go

```diff
@@ -389,6 +397,20 @@ func oidFromAuthCode(code string, config *v32.AzureADConfig, endpointURL string,
 	return authResult.IDToken.Oid, nil
 }
 
+func getMSGraphErrorData(err error) error {
+	switch err.(type) {
+	case *odataerrors.ODataError:
+		typed := err.(*odataerrors.ODataError)
+		errMsg := fmt.Sprintf("error: %v", typed.Error())
```

Comment:
This is nice, but you could simplify it.

Type-switches can have a declaration which is scoped to the cases.

You don't need `error: ` in front of errors, it's implied.

```go
func getMSGraphErrorData(err error) error {
    switch typed := err.(type) {
    case *odataerrors.ODataError:
        if terr := typed.GetError(); terr != nil {
            return fmt.Errorf("%s: code: %s, msg: %s", typed, *terr.GetCode(), *terr.GetMessage())
        }
        return typed
    default:
        return err 
    }   
}
```

Reply:
A `switch` with just one case is probably "overkill". Is it not easier a simple type assertion?

```go
func getMSGraphErrorData(err error) error {
    if odataErr, ok := err.(*odataerrors.ODataError); ok {
        if oErr := odataErr.GetError(); oErr != nil {
            return fmt.Errorf("%s: code: %s, msg: %s", odataErr, *oErr.GetCode(), *oErr.GetMessage())
        }
    }
    return err   
}
```

I don't have a strong opinion, I actually prefer the readability of the switch case.

Reply:
I changed the switch into a simple type assertion.